### PR TITLE
Add helper functions

### PR DESCRIPTION
**Pull request by Sourav-Patteri**
Updated code and resolved conversation. Added logic for other part numbers.

### CODE REVIEW

File: src/components/Logo.vue [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<template>
+    <div class="logo-container">
```
Comment (ksaswin) [inline src/components/Logo.vue]: Single quotes wherever possible, please.

File: src/components/Logo.vue [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      <img src="@/assets/UpwardorLogo.png" alt="Logo">
```
Comment (ksaswin) [inline src/components/Logo.vue]: Same here (single quotes).

File: src/components/QuoteGenerator.vue [hunk 3]
```diff
@@ -195,6 +197,55 @@ const displayDoorHeights = computed((): Array<string> => {
...
+
+  let frameTypeNum = chosenFrameSize.value === 'short' ? 0 : 1;  
+
+  const glazingKitPart = `GK15-1${frameTypeNum}${selectedPanelColor.value}-00`
+
+  // TODO: Decide and put in how to display the information. Tabular form?
+  console.log(`Your quote: Panel Part Numbers - ${panelPartNumber}. The Bottom Retainer part number is- ${bottomRetainerPart}. The Astragal is PL10-00005-01. The Glazing Kit is ${glazingKitPart}`);
+};
+
+const generatePanelPartNumber = (
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Declare such functions using the following syntax:
```ts
function generatePanelPartNumber(): string => {
  // ...
}
```

File: src/components/QuoteGenerator.vue [hunk 4]
```diff
@@ -195,6 +197,55 @@ const displayDoorHeights = computed((): Array<string> => {
 function updateExtrasList(extra: string): void {
   selectedExtras.value.push(extra);
 }
+
+function handleFormSubmit() : void {
+ 
+  // Call function to calculate quote
+  const panelPartNumber = generatePanelPartNumber(isPanelsOnlyOrder.value, selectedDoorHeight.value, selectedDoorWidth.value, doors.value, selectedStampPattern.value, selectedPanelColor.value);
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: The `ref` variables can be accessed within the `generatePanelPartNumber` function as well!

I would suggest simplifying the function definition for `generatePanelPartNumber` and update it here as:
```ts
const panelPartNumber = generatePanelPartNumber();
```

Or better, create a getter using `computed`:
```ts
// The panelPartNumber getter
const panelPartNumber = computed((): string => {
  // ...
});
```
The value can be accessed using `panelPartNumber.value`

File: src/components/QuoteGenerator.vue [hunk 5]
```diff
@@ -195,6 +197,55 @@ const displayDoorHeights = computed((): Array<string> => {
...
+const generatePanelPartNumber = (
+  bulkPanelsValue: boolean,
+  doorHeightValue: string | null,
+  doorWidthValue: string | null,
+  numberOfDoorsValue: number | null,
+  stampPatternValue: string | null,
+  panelColorValue: string | null
+): string => {
+  // Define the bulk panel prefix
+  const bulkPanelPrefix = bulkPanelsValue === true ? 'PN60' : 'PN65';
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Could be simplified:

```ts
const bulkPanelPrefix = bulkPanelsValue ? 'PN60' : 'PN65';
```

File: src/components/QuoteGenerator.vue [hunk 6]
```diff
@@ -195,6 +197,55 @@ const displayDoorHeights = computed((): Array<string> => {
...
+  const doorWidthSuffix = doorWidthValue.substring(0, 2);
+
+  const panelWidths = Object.keys(panelsTable[doorHeightSuffix] || {});
+
+  const partNumbers: string[] = [];
+
+  panelWidths.forEach((panelWidth) => {
+    const panelWidthSuffix = panelWidth.padStart(2, '0');
+    const panelCount = panelsTable[doorHeightSuffix][panelWidth];
+    if (panelCount > 0) {
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:220]: Style guide:
Leave an empty line before statements (like `if`, `return`, ...)

File: src/components/QuoteGenerator.vue [hunk 7]
```diff
@@ -195,6 +197,55 @@ const displayDoorHeights = computed((): Array<string> => {
...
+      const partNumber = `${bulkPanelPrefix}-${panelWidthSuffix}${stampPatternValue}${panelColorValue}-${doorWidthSuffix}00`;
+      partNumbers.push(partNumber);
+    }
+  });
+  return partNumbers.join(', ');
+
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue:228]: Could you remove the empty line here, please?

**Review by ksaswin — CHANGES_REQUESTED** [review]

File: src/components/QuoteGenerator.vue (hunk 3 above)
Comment (Sourav-Patteri) [reply src/components/QuoteGenerator.vue]: You mean- 
``function generatePanelPartNumber(): string {}``

File: src/components/QuoteGenerator.vue [hunk 8]
```diff
@@ -195,7 +197,49 @@ const displayDoorHeights = computed((): Array<string> => {
 function updateExtrasList(extra: string): void {
   selectedExtras.value.push(extra);
 }
+
+function handleFormSubmit() : void {
+ 
+    // Call function to calculate quote
+    const panelPartNumber = computed((): string => {
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Could you move this computed value outside `handleFormSubmit`, please?

```ts
const panelPartNumber = computed((): string => {
  //...
  return ‘value’;
});

function handleFormSubmit(): void {
  //...
console.log(`Your quote: Panel Part Numbers - ${panelPartNumber.value}….`); 
}
```

File: src/components/QuoteGenerator.vue [hunk 9]
```diff
@@ -195,7 +197,49 @@ const displayDoorHeights = computed((): Array<string> => {
...
+        partNumbers.push(partNumber);
+      }
+    });
+
+    return partNumbers.join(', ');
+  });
+
+  const bottomRetainerPart = bottomRetainerParts[selectedDoorWidth.value.substring(0, 2)];
+
+  let frameTypeNum = chosenFrameSize.value === 'short' ? 0 : 1;
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Prefer `const` if you don’t plan to change the value.

**Review by ksaswin — CHANGES_REQUESTED** [review]

File: src/components/QuoteGenerator.vue [hunk 10]
```diff
@@ -195,7 +197,49 @@ const displayDoorHeights = computed((): Array<string> => {
...
+    const doorWidthSuffix = selectedDoorWidth.value.substring(0, 2);
+
+    const panelWidths = Object.keys(panelsTable[doorHeightSuffix] || {});
+
+    const partNumbers: string[] = [];
+
+    panelWidths.forEach((panelWidth) => {
+      const panelWidthSuffix = panelWidth.padStart(2, '0');
+      const panelCount = panelsTable[doorHeightSuffix][panelWidth];
+
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Could you remove the trailing spaces here?

**Review by ksaswin — CHANGES_REQUESTED** [review]

File: src/components/QuoteGenerator.vue (hunk 9 above)
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Please remove the trailing spaces at the end of line 234.

File: src/components/QuoteGenerator.vue [hunk 11]
```diff
@@ -195,7 +197,49 @@ const displayDoorHeights = computed((): Array<string> => {
...
+    });
+
+    return partNumbers.join(', ');
+  });
+
+  const bottomRetainerPart = bottomRetainerParts[selectedDoorWidth.value.substring(0, 2)];
+
+  let frameTypeNum = chosenFrameSize.value === 'short' ? 0 : 1;  
+
+  const glazingKitPart = `GK15-1${frameTypeNum}${selectedPanelColor.value}-00`
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Missing semi-colon.

File: src/components/QuoteGenerator.vue [hunk 12]
```diff
@@ -195,7 +197,49 @@ const displayDoorHeights = computed((): Array<string> => {
...
+
+  // TODO: Decide and put in how to display the information. Tabular form?
+  console.log(`Your quote: Panel Part Numbers - ${panelPartNumber.value}. The Bottom Retainer part number is- ${bottomRetainerPart}. The Astragal is PL10-00005-01. The Glazing Kit is ${glazingKitPart}`);
+};
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: You don’t need a semi-colon here :)

File: src/components/QuoteGenerator.vue [hunk 13]
```diff
@@ -195,7 +197,49 @@ const displayDoorHeights = computed((): Array<string> => {
 function updateExtrasList(extra: string): void {
   selectedExtras.value.push(extra);
 }
+
+function handleFormSubmit() : void {
+ 
+    // Call function to calculate quote
+    const panelPartNumber = computed((): string => {
+
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Remove trailing spaces in lines 202 and 205 as well.

**Review by ksaswin — CHANGES_REQUESTED** [review]

File: src/components/QuoteGenerator.vue [hunk 14]
```diff
@@ -195,7 +197,49 @@ const displayDoorHeights = computed((): Array<string> => {
 function updateExtrasList(extra: string): void {
   selectedExtras.value.push(extra);
 }
+
+const panelPartNumber = computed((): string => {
+
+      // Define the bulk panel prefix
```
Comment (ksaswin) [inline src/components/QuoteGenerator.vue]: Fix the indentation in lines 203 -> 224 (use two blankspaces and not tabs to keep the consistensy across all editors out there).
And feel free to merge the PR.

**Review by ksaswin — APPROVED** [review]

**Pull request merged**